File: solutions/c/raindrops/1/raindrops.c
#include <stdio.h>
#include <string.h>
#include "raindrops.h"

void convert(char result[], int drops)
{
    if (drops % 3 == 0)
        snprintf(result, BUFFER_LENGTH, "%s", PLING);

    if (drops % 5 == 0)
        appendSound(result, PLANG);

    if (drops % 7 == 0)
        appendSound(result, PLONG);

    if (strlen(result) == 0)
        snprintf(result, BUFFER_LENGTH, "%d", drops);
}

void appendSound(char result[], char *sound)
{
    int currentSize = strlen(result);
    snprintf(result + currentSize, BUFFER_LENGTH - currentSize, "%s", sound);
}

File: solutions/ruby/raindrops/1/raindrops.rb
class Raindrops

  def self.convert(number)
    words = get_words(number)
    words.empty? ? number.to_s : words
  end

  private

  def self.get_words(number)
    pling = is_pling?(number) ? "Pling" : ""
    plang = is_plang?(number) ? "Plang" : ""
    plong = is_plong?(number) ? "Plong" : ""

    pling << plang << plong
  end

  def self.is_pling?(number)
    number % 3 == 0
  end

  def self.is_plang?(number)
    number % 5 == 0
  end

  def self.is_plong?(number)
    number % 7 == 0
  end

end
